src/test/versionbits_tests.cpp

```diff
@@ -242,81 +302,123 @@ BOOST_AUTO_TEST_CASE(versionbits_test)
     }
 }
 
-BOOST_AUTO_TEST_CASE(versionbits_computeblockversion)
+void check_computeblockversion(const Consensus::Params& params, Consensus::DeploymentPos dep)
 {
+    // This implicitly uses versionbitscache, so clear it every time
+    versionbitscache.Clear();
+
     // Check that ComputeBlockVersion will set the appropriate bit correctly
-    // on mainnet.
-    const auto chainParams = CreateChainParams(*m_node.args, CBaseChainParams::MAIN);
-    const Consensus::Params &mainnetParams = chainParams->GetConsensus();
 
     // Use the TESTDUMMY deployment for testing purposes.
-    int64_t bit = mainnetParams.vDeployments[Consensus::DEPLOYMENT_TESTDUMMY].bit;
-    int64_t nStartTime = mainnetParams.vDeployments[Consensus::DEPLOYMENT_TESTDUMMY].nStartTime;
-    int64_t nTimeout = mainnetParams.vDeployments[Consensus::DEPLOYMENT_TESTDUMMY].nTimeout;
+    int64_t bit = params.vDeployments[Consensus::DEPLOYMENT_TESTDUMMY].bit;
```

Comment:
Is there a reason that these are `vDeployments[Consensus::DEPLOYMENT_TESTDUMMY]` and not `vDeployments[dep]`?

Reply:
Nope, must have missed replacing all those in some rebase/redraft or another.